device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/DeviceIO.java

```diff
@@ -146,7 +146,7 @@ void open() throws IOException
     {
         synchronized (this.stateLock)
         {
-            if (this.state == IotHubConnectionStatus.CONNECTED || this.state == IotHubConnectionStatus.DISCONNECTED_RETRYING)
+            if (this.isOpen())
```

Comment:
Do we already have e2e/UT reflecting this behavior?

Reply:
Just added a unit test for this

Reply:
Some of our e2e tests were already hitting this issue sometimes which is why I caught this in the first place

Reply:
👍